presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/EvaluateZeroLimit.java

```diff
@@ -27,10 +28,18 @@
 public class EvaluateZeroLimit
     implements Rule
 {
+    private static final Pattern PATTERN = Pattern.node(LimitNode.class);
+
+    @Override
+    public Pattern getPattern()
+    {
+        return PATTERN;
+    }
+
     @Override
     public Optional<PlanNode> apply(PlanNode node, Lookup lookup, PlanNodeIdAllocator idAllocator, SymbolAllocator symbolAllocator)
     {
-        if (!(node instanceof LimitNode)) {
+        if (!PATTERN.matches(node)) {
```

Comment:
* Do i still need to test this? Once I expressed by basic expectations in `getPattern`, I should not need to re-test them here.
* with this gone, I don't need to define `PATTERN` field at all, do I?